src/supernova/mod.rs

```diff
@@ -129,139 +113,226 @@ where
 where
   G1: Group<Base = <G2 as Group>::Scalar>,
   G2: Group<Base = <G1 as Group>::Scalar>,
   C1: EnforcingStepCircuit<G1::Scalar>,
   C2: EnforcingStepCircuit<G2::Scalar>,
 {
-  _phantom: PhantomData<C1>,
-  augmented_circuit_index: usize,
-  c_secondary: C2,
-  params: PublicParams<G1, G2>,
+  /// The internal public params
+  pub circuit_params: Vec<CircuitParams<G1, G2>>,
+
+  ck_primary: CommitmentKey<G1>, // This is shared between all circuit params
+
+  F_arity_secondary: usize,
+  ro_consts_secondary: ROConstants<G2>,
+  ro_consts_circuit_secondary: ROConstantsCircuit<G1>,
+  ck_secondary: CommitmentKey<G2>,
+  r1cs_shape_secondary: R1CSShape<G2>,
+  augmented_circuit_params_secondary: SuperNovaAugmentedCircuitParams,
+
+  /// Digest constructed from these `RunningClaim`s' parameters
+  // Once we serialize this struct, it's important to add this
+  // annotaiton to the digest field for it to be treated properly: #[serde(skip, default = "OnceCell::new")]
+  digest: OnceCell<G1::Scalar>,
+  _p: PhantomData<(C1, C2)>,
 }
 
-/// Indexed list of `RunningClaim`s representing an NIVC computation in process.
-pub struct RunningClaims<G1, G2, C1, C2>
+/// Auxillary [PublicParams] information about the commitment keys and
+/// secondary circuit. This is used as a helper struct when reconstructing
+/// [PublicParams] downstream in lurk.
+#[derive(Clone, PartialEq, Serialize, Deserialize, Abomonation)]
+#[serde(bound = "")]
+#[abomonation_bounds(
+where
+  G1: Group<Base = <G2 as Group>::Scalar>,
+  G2: Group<Base = <G1 as Group>::Scalar>,
+  <G1::Scalar as PrimeField>::Repr: Abomonation,
+  <G2::Scalar as PrimeField>::Repr: Abomonation,
+)]
+pub struct AuxParams<G1, G2>
 where
   G1: Group<Base = <G2 as Group>::Scalar>,
   G2: Group<Base = <G1 as Group>::Scalar>,
-  C1: StepCircuit<G1::Scalar>,
-  C2: StepCircuit<G2::Scalar>,
 {
-  /// Indexed `RunningClaim`s
-  claims: Vec<RunningClaim<G1, G2, C1, C2>>,
-  /// Digest constructed from these `RunningClaim`s' parameters
-  // Once we serialize this struct, it's important to add this
-  // annotaiton to the digest field for it to be treated properly: #[serde(skip, default = "OnceCell::new")]
-  digest: OnceCell<G1::Scalar>,
+  ck_primary: CommitmentKey<G1>, // This is shared between all circuit params
+
+  F_arity_secondary: usize,
+  ro_consts_secondary: ROConstants<G2>,
+  ro_consts_circuit_secondary: ROConstantsCircuit<G1>,
+  ck_secondary: CommitmentKey<G2>,
+  r1cs_shape_secondary: R1CSShape<G2>,
+  augmented_circuit_params_secondary: SuperNovaAugmentedCircuitParams,
+
+  #[abomonate_with(<G1::Scalar as PrimeField>::Repr)]
+  digest: G1::Scalar,
 }
 
-impl<G1, G2, C1, C2> Index<usize> for RunningClaims<G1, G2, C1, C2>
+impl<G1, G2, C1, C2> Index<usize> for PublicParams<G1, G2, C1, C2>
 where
   G1: Group<Base = <G2 as Group>::Scalar>,
   G2: Group<Base = <G1 as Group>::Scalar>,
-  C1: StepCircuit<G1::Scalar>,
-  C2: StepCircuit<G2::Scalar>,
+  C1: EnforcingStepCircuit<G1::Scalar>,
```

Comment:
This extension of bounds to an `EnforcingStepCircuit` should not be needed, specially if you don't use `enforcing_synthesis`. 

The way this works is when using the method `synthesize` on any `SuperNovaAugmentedCircuit`, Rust will find there is a requirement of the circuit being an `EnforcingStepCircuit`, and seeking that instance, will find the blanket implementation of this [extension trait](http://xion.io/post/code/rust-extension-traits.html):
https://github.com/lurk-lab/arecibo/blob/874c06c0203a0977cd3fc2f5848ca9649a1b988f/src/supernova/circuit.rs#L422-L423

Promoting requirements to an `EnforcingStepCircuit` is just noise. Consider the following diff please:
https://gist.github.com/huitseeker/74a4c4ee55e8b15459265236b6703c9e